src/lib/es5.d.ts

```diff
@@ -1229,6 +1229,12 @@ interface ReadonlyArray<T> {
      * @param thisArg An object to which the this keyword can refer in the predicate function. If thisArg is omitted, undefined is used as the this value.
      */
     filter<S extends T>(predicate: (value: T, index: number, array: readonly T[]) => value is S, thisArg?: any): S[];
+    /**
+     * Filters out nullish values from the array. Used when `Boolean` is passed as the argument to filter.
+     * @param predicate the `Boolean` constructor, which validates the truthiness of the value being mapped over.
+     * @param thisArg An object to which the this keyword can refer in the predicate function. If thisArg is omitted, undefined is used as the this value.
+     */
+    filter<S extends T>(predicate: BooleanConstructor, thisArg?: any): NonNullable<S>[];

```

Comment:
This looks like a hack solution. Is it to improve how ts infer the return type of a function so it can infer a type predicate?

```ts
arr.filter(x => typeof x === 'number')
// T[].filter<(x: T) => x is number>

arr.filter(x => x instanceof Q)
// T[].filter<(x: T) => x is Q>

arr.filter(x => x.kind === 1)
// ({ kind: 1, data: T } | { kind: 2, data: Q })[].filter<(x: ...) => x is { kind: 1, data: T }>
```

Reply:
No, this has nothing to do with inferring type predicates from function expressions, it’s just so you can use `.filter(Boolean)`.  See #50387.